src/basic/user-util.h

```diff
@@ -42,6 +42,9 @@ char* gid_to_name(gid_t gid);
 int in_gid(gid_t gid);
 int in_group(const char *name);
 
+size_t merge_gid_lists(const gid_t *list1, size_t size1, const gid_t *list2, size_t size2, gid_t *result);
+int get_current_gids(gid_t** gids);
```

Comment:
I'd prefer to call this `getgroups_alloc`. Then it is clear that it is a wrapper around `getgroups`. We follow this pattern with a number of other functions that need an output buffer.